src/backend/joanie/lms_handler/backends/openedx.py

```diff
@@ -0,0 +1,117 @@
+"""
+Backend to connect Joanie to OpenEdX LMS
+"""
+import json
+import logging
+import re
+
+import requests
+from requests.auth import AuthBase
+
+from .base import BaseLMSBackend
+
+logger = logging.getLogger(__name__)
+
+
+class OpenEdXTokenAuth(AuthBase):
+    """Attach HTTP token authentication header to the given Request object."""
+
+    def __init__(self, token):
+        """Set-up token value in the instance."""
+        self.token = token
+
+    def __call__(self, request):
+        """Modify and return the request."""
+        request.headers.update(
+            {
+                "Content-Type": "application/json",
+                "X-Edx-Api-Key": self.token,
+            }
+        )
+
+        return request
+
+
+class TokenAPIClient(requests.Session):
+    """
+    A class `request.Session` that automatically authenticates agains OpenEdX's preferred
+    authentication method up to Dogwood, given a secret token.
+
+    For more usage details, see documentation of the class `requests.Session` object:
+    https://requests.readthedocs.io/en/master/user/advanced/#session-objects
+    """
+
+    def __init__(self, token, *args, **kwargs):
+        """Extending the session object by setting the authentication token."""
+        super().__init__(*args, **kwargs)
+        self.auth = OpenEdXTokenAuth(token)
+
+
+class OpenEdXLMSBackend(BaseLMSBackend):
+    @property
+    def api_client(self):
+        """Instantiate and return an OpenEdX token API client."""
+        return TokenAPIClient(self.configuration["API_TOKEN"])
+
+    def extract_course_id(self, resource_link):
+        """Extract the LMS course id from the course run url."""
+        return re.match(self.configuration["COURSE_REGEX"], resource_link).group(
+            "course_id"
+        )
+
+    def parse_course_key(self, course_key):
+        """
+        Split an OpenEdX course key by organization, course and course run codes.
+
+        We first try splitting the key as a version 1 key (course-v1:org+course+run)
```

Comment:
it can be `course-v[0-9]` no ? Not only `course-v1`

Reply:
You are right.

>Old course_id format: {school}/{course}/{run}
>Example: edX/DemoX/Demo_Course
>
>New course_id format: course-v{version number}:{school}+{course}+{run}
>Example: course-v1:MITx+6.00.2x_3+1T2015



https://openedx.atlassian.net/wiki/spaces/AC/pages/28967341/edX+Glossary

Reply:
If you want to see how we manage it in marsha: https://github.com/openfun/marsha/blob/d0da96eefbf2e1104e893d29fad7567d689883c6/src/backend/marsha/core/lti/__init__.py#L204-L211